pkg/plugin/processor/plugin.go

```diff
@@ -0,0 +1,147 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate mockgen -destination=mock/plugin.go -package=mock -mock_names=ProcessorPlugin=ProcessorPlugin . ProcessorPlugin
+
+package processor
+
+import (
+	"context"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+)
+
+// ProcessorPlugin receives records, manipulates them and returns back the processed
+// records.
+//
+//nolint:revive // This interface will be moved into the SDK
+type ProcessorPlugin interface {
```

Comment:
As per the conversation we had earlier: https://github.com/ConduitIO/conduit-processor-sdk/pull/1

Reply:
The PR above was merged.